src/main/java/backend/Logic.java

```diff
@@ -204,14 +199,21 @@ public Model getRepo(String repoId) {
      * @return True if label replacement on GitHub was a success, false otherwise.
```

Comment:
The description need to be complete. e.g.
`If the given set of labels are different from the ones in the issue, updates the issue object and the UI with newLabels. Pushes the labels to the GitHub server irrespective of whether the labels are changed. Reason: so that this method can be called again to resolve persistent inconsistencies between the repo's state and the UI's state. Remote update is done after the local update, to avoid the lag between the user action and UI response.`

Now, if the last sentence means this is an 'eager' update, then you should put the 'eager' back into the method name.

More importantly, see how messy the method description is. That's a sign that this method is not well defined. What if we remove the equivalence test?  Then the description becomes cleaner.

`Replaces existing labels with new labels in the issue object, the UI, and the server, in that order. Server update is done after the local update to reduce the lag between the user action and the UI response.`